src/lightning_lite/lite.py

```diff
@@ -517,10 +582,37 @@ def _validate_setup(model: nn.Module, optimizers: Sequence[Optimizer]) -> None:
         if any(isinstance(opt, _LiteOptimizer) for opt in optimizers):
             raise ValueError("An optimizer should be passed only once to the `setup` method.")
 
+    def _validate_setup_model(self, model: nn.Module) -> None:
+        if isinstance(model, _LiteModule):
+            raise ValueError("A model should be passed only once to the `setup_model` method.")
+
+        if isinstance(self._strategy, (DDPShardedStrategy, DDPSpawnShardedStrategy)):
```

Comment:
It may seems so right now, but actually the _Sharded type does not determine whether a strategy supports being setup in a particular order. You see, there are two examples

- DDPShardedStrategy: Needs joint setup of optimizer and model
- FSDPStrategy: Needs setup model first before optimizer

both are of type _Sharded :)

Reply:
should we then have a flag on the strategy indicating that so that we don't have to maintain that list here?

Reply:
For now this should be fine. I would avoid adding new attributes on the strategies unless absolutely necessary.

The strategies anyway need to handle this in the respective methods by producing an error if you call the wrong methods(s), e.g., FSDP gives an error if you call `strategy.setup_module_and_optimizer()`. But the error message will reference the API of the strategy itself. That's why I added custom error messages here in Lite so that we can reference the lite.x usage to better inform the user what they should do.